app/facades/place_facade.rb

```diff
@@ -13,4 +13,13 @@ def find_place
     photo = place_service.find_photo(place_ref)
     place_item = Place.new(place[:candidates].first[:name], photo.env[:response_headers][:location])
   end
+
+  def find_random_restaurants
+    service = place_service.random_restaurants(@query)
+    places = service[:results].map do |place|
+      place_ref = place[:photos][0][:photo_reference]
+      photo = place_service.find_photo(place_ref)
+      place_item = Place.new(place[:name], photo.env[:response_headers][:location])
```

Comment:
This is fairly minor but because you call `place_service` multiple times in this method, I would change the `place_service` method on line 6 to:
```ruby
def place_service
  @place_service ||= PlaceService.new
end
```

This way you don't need to re-initialize for every call. You probably won't notice much of a difference in practice but it can make calls more efficient if you were dealing with higher user load.

Reply:
done!